tests/compile-fail/intrinsics/float_to_int_64_too_big4.rs

```diff
@@ -0,0 +1,10 @@
+#![feature(intrinsics)]
+
+// Directly call intrinsic to avoid debug assertions in libstd
+extern "rust-intrinsic" {
+    fn float_to_int_unchecked<Float: Copy, Int: Copy>(value: Float) -> Int;
+}
+
+fn main() {
+    unsafe { float_to_int_unchecked::<f64, u128>(340282366920938463463374607431768211455.0f64); } //~ ERROR: cannot be represented in target type `u128`
```

Comment:
I *think* this test is correct, but it looks misleading or confusing. 340282366920938463463374607431768211455 (the integer) is `u128::MAX`, so at first glance the conversion should succeed. But the `340282366920938463463374607431768211455.0f64` float is in fact not equal to that integer. That integer cannot be represented exactly in f64, so the Rust parser presumably rounds to the nearest representable float which is one more, 2^128, and does indeed overflow conversion to u128.

(I’m using Python to figure this out since it has infinite-capacity integers and `f64` floats. `a = 340282366920938463463374607431768211455; (hex(a), int(float(a)) - a, a.bit_length())` prints `('0xffffffffffffffffffffffffffffffff', 1, 128)`.)

But the point of these tests is not to test the Rust parser. I feel it’d be better to pick a decimal representation of the float that would, if precision and range were infinite, have the same numerical value as the actual f64 value does.

In fact it’d be even better if the source code of test cases could contain something closer to the memory representation of `f64`. Hex float syntax exists, but not in Rust literals. I don’t know if this would be worth using a library for parsing it from strings. I see some other tests use `f32::from_bits` with hex integers, but that’s much harder for humans to read.

Reply:
FWIW, that number here and these `from_bits` constants are all derived from https://github.com/WebAssembly/testsuite/blob/master/conversions.wast.

I could change this to `u128::MAX as f64`, if you think that's better.